ods_ci/tests/Tests/100__deploy/130__operators/130__rhods_operator/139__trusted_ca_bundles.robot

```diff
@@ -0,0 +1,155 @@
+
+    Log    message=Restore DSCI to original state    level=INFO
+    Set Custom CA Bundle Value In DSCI   ''    ${RHOAI_OPERATOR_NS}
+    Set Trusted CA Bundle Management State    Managed    ${RHOAI_OPERATOR_NS}
+    Delete Namespace From Openshift    ${TEST_NS}
+
+
+*** Keywords ***
+Suite Setup
+    [Documentation]    Suite Setup
+    Log    message=Suite Setup.    level=INFO
+    RHOSi Setup
+
+Suite Teardown
+    [Documentation]    Suite Teardown
+    Log    message=Suite Teardown.    level=INFO
+    RHOSi Teardown
+
+Is Operator Available
+    [Documentation]    Checks if operator is available
+    [Arguments]    ${operator_name}    ${namespace}
+    ${rc}=    Run And Return Rc
+    ...    oc wait --timeout=2m --for condition=available -n ${namespace} deploy/${operator_name}
+    Should Be Equal As Integers    ${rc}     ${0}    msg=${operator_name} Not Available
+
+Is Resource Present
+    [Documentation]    Check If Resource Is Present In Namespace
+    [Arguments]       ${resource}     ${resource_name}    ${namespace}    ${expected_result}
+    ${rc}=     Run And Return Rc
+    ...  oc get ${resource} ${resource_name} -n ${namespace}
+    Should Be Equal    "${rc}"    "${expected_result}"    msg=${resource} does not exist in ${namespace}
+
+Is DSCI In Ready State
+    [Documentation]    Checks that DSCI Reconciled Succesfully
+    [Arguments]    ${dsci}    ${namespace}
+    ${rc}=    Run And Return Rc
+    ...    oc wait --timeout=3m --for jsonpath='{.status.conditions[].reason}'=ReconcileCompleted -n ${namespace} dsci ${dsci}
+    Should Be Equal As Integers    ${rc}     ${0}    msg=${dsci} not in Ready state
+
+Check ConfigMap Contains CA Bundle
+    [Documentation]    Checks that ConfigMap contains CA Bundle
+    [Arguments]    ${config_map}    ${ca_bundle_name}    ${namespace}
+    ${rc}=     Run And Return Rc
+    ...    oc get configmap ${config_map} -n ${namespace} -o yaml | grep ${ca_bundle_name}
+    Should Be Equal    "${rc}"    "0"    msg=${TRUSTED_CA_BUNDLE_CONFIGMAP} does not contain CA bundle ${ca_bundle_name}
+
+Is CA Bundle Value Present
+    [Documentation]    Check if the ConfigtMap contains Custom CA Bundle value
+    [Arguments]    ${config_map}    ${custom_ca_bundle_value}    ${namespace}        ${expected_result}
+    ${rc}=     Run And Return Rc
+    ...    oc get configmap ${config_map} -n ${namespace} -o yaml | grep ${custom_ca_bundle_value}
+    Should Be Equal As Integers    ${rc}    ${expected_result}
+
+Create Namespace In Openshift
+    [Documentation]    Create a new namespace if it does not already exist
+    [Arguments]    ${namespace}
+    ${rc}=     Run And Return Rc    oc get project ${namespace}
+    IF    ${rc} != 0
+        ${rc}=     Run And Return Rc    oc new-project ${namespace}
+        Should Be Equal    "${rc}"    "0"    msg=Failed to create namespace ${namespace}
+    END
+
+Delete Namespace From Openshift
```

Comment:
I think this keyword exists in ods-ci

Reply:
Found similar, but slightly different implementation. Will continue to investigate for near future PR.